compio-io/src/write/mod.rs

```diff
@@ -181,15 +159,45 @@ macro_rules! impl_write_at {
 impl_write_at!(@ptr &mut A, Box<A>);
 impl_write_at!(@slice [u8], const LEN => [u8; LEN]);
 
-impl AsyncWriteAt for Vec<u8> {
+impl<#[cfg(feature = "allocator_api")] A: Allocator> AsyncWriteAt for vec_alloc!(u8, A) {
```

Comment:
Add some documentation for this special implementation, especially the 0-filled extended memory when pos > len.